File: ABC/B/254/254.py
if __name__ == '__main__':
    vec = [1]
    
    n = int(input())
    print(1)

    for i in range(1, n):
        buf = []
        buf.append(1)
        for j in range(i-1):
            buf.append(vec[j] + vec[j+1])
        buf.append(1)
        vec = buf
        for j in range(i):
            print(str(vec[j]), end = " ")
        print(vec[i])

File: ABC/B/254/254.cpp
#include<iostream>
#include<vector>
using namespace std;

int n;
vector<int> vec;

int main(){
    cin >> n;
    cout << 1 << endl;

    vec.push_back(1);
    for(int i=0; i<n-1; i++){
        vector<int> buf;
        buf.push_back(1);
        for(int j=0; j<i; j++){
            buf.push_back(vec.at(j) + vec.at(j+1));
        }
        buf.push_back(1);
        vec = buf;
        for(int j=0; j<i+1; j++){
            cout << vec.at(j) << ' ';
        }
        cout << vec.at(i+1) << endl;
    }
}   